### PR TITLE
39-feature-add-field-niedostępny-to-the-post-view

### PR DESCRIPTION
**Pull request by Irysek707**
In this PR:
1. Added $awaybit function to the functions_post.php.
2. Added away status to the postbit view.
3. Added styling to the status_away class.
4. Fixed postbit_classic so it looks the same as postbit.

### CODE REVIEW

File: templates/4_crystallicum_dark_templates/postbit/postbit_classic.html [hunk 1]
```diff
@@ -1,45 +1,94 @@
...
-			</span>
-	</div>
-	<div class="author_statistics">
-		{$post['user_details']}
-	</div>
-</div>
-<div class="post_content">
+<script>
+	document.addEventListener('DOMContentLoaded', function() {
+	  // Get the author avatar element
```
Comment (MysteriousGreenBean) [inline templates/4_crystallicum_dark_templates/postbit/postbit_classic.html:3]: I know there's something like CSS variables that should allow you get this effect without JS.
...well, at least I THINK so.
Here's solution by chatGPT, so you can test it, but I think this (...or something like this) should be possibe:
```CSS
.author_avatar {
  --avatar-height: calc(100% + 10px); /* Adding 10px as per your original logic */
  height: auto; /* Ensures the avatar adjusts its height naturally */
}

.post_author {
  min-height: var(--avatar-height);
}
```

File: templates/4_crystallicum_dark_templates/postbit/postbit_classic.html [hunk 2]
```diff
@@ -1,45 +1,94 @@
...
+  
+		// Set the min-height of the post author
+		postAuthor.style.minHeight = (avatarHeight + 10) + 'px';
+	  }
+	});
+  </script>
+  
+  <!--- Mechanika --->
+  <script>
+	document.addEventListener('DOMContentLoaded', function() {
```
Comment (MysteriousGreenBean) [inline templates/4_crystallicum_dark_templates/postbit/postbit_classic.html:22]: I'd really like us not to end up having hundreds of events registered for DOMContentLoaded. Why not just make it a function and call said function in onClick event for specific elements? That's something that can be made via template itself

File: website/inc/functions_post.php [hunk 3]
```diff
@@ -234,6 +234,56 @@ function build_postbit($post, $post_type=0)
 		unset($usertitle, $cached_titles);
 	}
 
+	// Add this part where the user's data is being fetched
+	$user = get_user($post['uid']); // Assuming user data is not already loaded
+
+	$awaybit = '';
+if ($post['away'] == 1 && $mybb->settings['allowaway'] != 0) {
```
Comment (MysteriousGreenBean) [inline website/inc/functions_post.php:241]: NIT: `$mybb->settings['allowaway'] == 1`

File: templates/4_crystallicum_dark_templates/postbit/postbit_classic.html (hunk 2 above)
Comment (MysteriousGreenBean) [reply templates/4_crystallicum_dark_templates/postbit/postbit_classic.html]: Done

File: templates/4_crystallicum_dark_templates/postbit/postbit_classic.html (hunk 1 above)
Comment (MysteriousGreenBean) [reply templates/4_crystallicum_dark_templates/postbit/postbit_classic.html]: It seems to be working without it - you can remove the script, we'll worry about it when issue happens again :D

File: website/inc/functions_post.php (hunk 3 above)
Comment (Irysek707) [reply website/inc/functions_post.php:241]: Fixed.

File: website/inc/functions_post.php [hunk 4]
```diff
@@ -234,6 +234,56 @@ function build_postbit($post, $post_type=0)
...
+        if ($returnmkdate < TIME_NOW) {
+            $db->update_query('users', array('away' => '0', 'awaydate' => '0', 'returndate' => '', 'awayreason' => ''), 'uid=\''.(int)$post['uid'].'\'');
+            $post['away'] = 0;
+        }
+    }
+
+    // Check if still away after updates
+    if ($post['away'] == 1) {
+        // You may need a new template bit for postbit away message
+        eval("\$awaybit = \"".$templates->get("postbit_away")."\";");
```
[the remark this answers is not in the document]
Comment (Irysek707) [reply website/inc/functions_post.php:277]: Can't be changed :(

**Review by MysteriousGreenBean — APPROVED** [review]

**Pull request merged**